src/java/org/apache/cassandra/streaming/StreamDeserializingTask.java

```diff
@@ -109,6 +110,10 @@ public StreamSession deriveSession(StreamMessage message)
         // Attach this channel to the session: this only happens upon receiving the first init message as a follower;
         // in all other cases, no new control channel will be added, as the proper control channel will be already attached.
         streamSession.attachInbound(channel);
+
+        if (streamSession.getStreamOperation() == StreamOperation.BULK_LOAD && !Guardrails.bulkLoadEnabled.isEnabled(null))
+            throw new StreamReceiveException(streamSession, new RuntimeException("Bulk load is disabled"));
```

Comment:
This isn't a proper usage of guardrails. They should be checked with `ensureEnabled()`, so they throw the formatted `GuardrailViolatedException` defined in their constructor. That would also trigger other associated behaviours, like emitting a log message and a diagnostic event.

Instead, the usage of `EnableFlag#isEnabled` is ignoring all that and using it as a regular boolean config property.

An issue here might be that we don't have a `ClientState`, and guardrails don't throw an exception but a warning on `null` client states: https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/db/guardrails/Guardrail.java#L139-L140

I think we could add a new `Guardrail#throwOnNullClientState()` method to instruct guardrails to throw an exception when triggered with a null `ClientState`. We usually don't throw on null client states to not break background processes such as compaction, but it can make sense for unauthenticated tools like this one.

Reply:
Made the change please review it again. Thanks.

Reply:
@adelapena thank you for catching this! It is easy to overlook the difference between `isEnabled` and `ensureEnabled` without deeper look when doing a review just from a browser. On face value it looks like it does the correct job, even returns `boolean`.